checkbox-ng/plainbox/impl/execution.py

```diff
@@ -374,7 +374,9 @@ def get_record_path_for_job(self, job):
 
     def send_signal(self, signal, target_user):
         if not target_user:
-            os.kill(self._running_jobs_pid, signal)
+            if self._running_jobs_pid:
+                return os.kill(self._running_jobs_pid, signal)
```

Comment:
could this happen in the `else` clause, that is, if there is a `target_user`? Should the existence of `self._running_jobs_pid` be checked regardless of this? (I don't have an answer to this question, but I saw that `self._running_jobs_pid` is being used below when running a `sudo` command to kill the process, so...

Reply:
in the `else` block we don't crash on failed kills, so we don't have to. But if we want to be courteous then yes.